.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,41 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '20 19 * * 6'
+
+jobs:
+  analyze:
+    name: CodeQL
+    runs-on: windows-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```

Comment:
should we add typescript here? notice that ts is also supported.

Reply:
You can reference this issue: https://github.com/github/codeql-action/issues/365